pkg/webhooks/provisioning/v1alpha1/defaulter.go

```diff
@@ -49,20 +53,39 @@ func (v *Defaulter) Handle(ctx context.Context, req admission.Request) admission
 	if err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
-	v.applyDefaults(&provisioner.Spec)
+	if err := v.applyDefaults(ctx, &provisioner.Spec); err != nil {
+		return admission.Errored(0, fmt.Errorf("applying defaults to provisioner, %w", err))
+	}
+
 	marshaled, err := json.Marshal(provisioner)
 	if err != nil {
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaled)
 }
 
-func (v *Defaulter) applyDefaults(spec *provisioning.ProvisionerSpec) {
-	v.defaultTTL(spec)
+func (v *Defaulter) applyDefaults(ctx context.Context, spec *provisioning.ProvisionerSpec) error {
+	return functional.ValidateAll(
+		func() error { return v.defaultTTL(spec) },
+		func() error { return v.defaultCapacityType(ctx, spec) },
+	)
 }
 
-func (v *Defaulter) defaultTTL(spec *provisioning.ProvisionerSpec) {
+func (v *Defaulter) defaultTTL(spec *provisioning.ProvisionerSpec) error {
 	if spec.TTLSeconds == nil {
 		spec.TTLSeconds = ptr.Int32(300)
 	}
+	return nil
+}
+
+func (v *Defaulter) defaultCapacityType(ctx context.Context, spec *provisioning.ProvisionerSpec) error {
+	if spec.CapacityType != nil {
+		return nil
+	}
+	capacityType, err := v.CloudProvider.CapacityFor(spec).DefaultCapacityType(ctx)
```

Comment:
Do we need to default anything into the provisioner at all? Can we simply let unspecified allow the controller to do what it will?

Reply:
So the options are:
 1) Default explicitly by calling a DefaultCapacityType()
 2) Default based on first entry in GetCapacityTypes()
 3) Or allow for empty capacityType and let the cloud provider decide 

I'm not sure which one I like the best... I like the explicit nature of 1, 2 requires less fluff in the cloudprovider interface. 

3 seems like it could be confusing. Especially since we've had discussions on spot being default or on-demand being default... so cloudproviders deciding seems too unpredictable.

Reply:
I'd say something like #3 makes the most sense. The default should be on-demand and cloud providers should default to on-demand.